### PR TITLE
Initialize the `worldPoseDirty` flag in Collision.cc.

### PR DESCRIPTION
**Pull request by zmichaels11**
This potentially resolves a rarely occuring issue in long-running runs of gzserver.
The uninitialized value was found by running Gazebo in Valgrind.

Signed-off-by: Zachary Michaels <zmichaels11@gmail.com>>

### CODE REVIEW

**Comment by emersonknapp** [comment]
@chapulina friendly ping for review - can we assign a maintainer reviewer for this PR?

**Review by ahcorde — APPROVED** [review]

**Comment by emersonknapp** [comment]
Thanks @ahcorde ! We're new to gazebo contributions - are the Checks expected to fail right now? Is there a next CI step that's needed after reviewer approval?

**Review by scpeters — APPROVED** [review]
our CI has a lot of pre-existing test failures; your changes look good to me

**Pull request merged**
